openfisca_uk/variables/finance/benefit/family/universal_credit.py

```diff
@@ -138,43 +363,49 @@ def formula(benunit, period, parameters):
                 ["income_tax", "national_insurance"],
             ),
         )
-        housing_element = benunit("UC_eligible_rent", period)
+        housing = benunit("UC_housing_costs_element", period)
         earnings_disregard = (
-            housing_element > 0
+            housing > 0
         ) * UC.means_test.earn_disregard_with_housing * MONTHS_IN_YEAR + (
-            housing_element == 0
+            housing == 0
         ) * UC.means_test.earn_disregard * MONTHS_IN_YEAR
-        earnings_reduction = max_(0, earned_income - earnings_disregard)
-        reduction = max_(
-            0,
-            unearned_income
-            + UC.means_test.reduction_rate * earnings_reduction,
-        )
-        return reduction
+        return max_(0, earned_income - earnings_disregard)
 
 
-class universal_credit_eligible(Variable):
-    value_type = bool
+class UC_unearned_income(Variable):
+    value_type = float
     entity = BenUnit
-    label = u"Whether eligible for Universal Credit"
+    label = u"UC unearned income"
     definition_period = YEAR
 
     def formula(benunit, period, parameters):
-        DISQUALIFYING_BENEFITS = [
-            "working_tax_credit",
-            "child_tax_credit",
-            "housing_benefit",
-            "income_support",
-            "ESA_income",
-            "JSA_income",
-        ]
-        disqualifying = add(benunit, period, DISQUALIFYING_BENEFITS) + aggr(
-            benunit, period, ["SDA"]
+        return aggr(
+            benunit,
+            period,
+            [
+                "carers_allowance",
+                "JSA_contrib",
+                "state_pension",
+                "pension_income",
+            ],
         )
-        eligible = (disqualifying == 0) * (
-            not_(benunit.min(benunit.members("is_SP_age", period)))
+
+
+class UC_income_reduction(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"Reduction from income for Universal Credit"
+    definition_period = YEAR
+
+    def formula(benunit, period, parameters):
+        UC = parameters(period).benefit.universal_credit
+        reduction = max_(
+            0,
+            benunit("UC_unearned_income", period)
+            + UC.means_test.reduction_rate
```

Comment:
could you make this a new variable e.g. `reduction_from_earned_income`?

Reply:
Could do... might leave for future now though if it's not immediately needed